### PR TITLE
feat: supports the function of vocabulary

### PR DESCRIPTION
**Pull request by Jambo2018**
1. add indexdb to collect words queried 
2. update query frequency for words 
3. add Random Review Mode of vocabulary 
4. support the export of collection in csv 
5. add article customization to deepen the memory of words 
6. optimize the button group of the input

### CODE REVIEW

File: src/content_script/PopupCard.tsx [hunk 1]
```diff
@@ -471,6 +488,30 @@ export function PopupCard(props: IPopupCardProps) {
     const [originalText, setOriginalText] = useState(props.text)
     const [translatedText, setTranslatedText] = useState('')
     const [translatedLines, setTranslatedLines] = useState<string[]>([])
+    const [wordMode, setWordMode] = useState<boolean>(false)
+    const [collectd, setColleced] = useState<boolean>(false)
```
Comment (yetone) [inline src/content_script/PopupCard.tsx:492]:
```suggestion
    const [isCollected, setIsCollected] = useState(false)
```

File: src/content_script/PopupCard.tsx [hunk 2]
```diff
@@ -471,6 +488,30 @@ export function PopupCard(props: IPopupCardProps) {
...
+    const [collectTotal, updateTotal] = useState<number>(0)
```
Comment (yetone) [inline src/content_script/PopupCard.tsx:493]:
```suggestion
    const [collectTotal, setCollectTotal] = useState(0)
```

File: src/content_script/PopupCard.tsx [hunk 3]
```diff
@@ -375,9 +392,9 @@ export function PopupCard(props: IPopupCardProps) {
     const editorRef = useRef<HTMLTextAreaElement>(null)
     const isCompositing = useRef(false)
     const [selectedWord, setSelectedWord] = useState('')
-
+    const [vocabularyType, setVocabularyType] = useState<'hide' | 'vocabulary' | 'article'>('hide')
     const highlightRef = useRef<HighlightInTextarea | null>(null)
-
+    const [showMoreBtn, setShowMore] = useState<boolean>(false)
```
Comment (yetone) [inline src/content_script/PopupCard.tsx:397]:
```suggestion
    const [showMoreBtn, setShowMoreBtn] = useState(false)
```

**Comment by yetone** [comment]
Very COOL!

**Comment by Jambo2018** [comment]
Sorry,I tried to use it to display the word list, but I gave up later and
forgot to delet the dependency info.😬

yetone ***@***.***> 于2023年3月30日周四 15:25写道：

File: package.json [hunk 4]
```diff
@@ -52,6 +54,7 @@
     "languagedetect": "^2.0.0",
     "lru-cache": "^8.0.0",
     "rc-field-form": "^1.27.4",
+    "rc-table": "^7.31.1",
```
[the remark this answers is not in the document]
Comment (Jambo2018) [reply package.json]: its been deleted

**Comment by yetone** [comment]
I just tested it locally, so cool!!!

**Pull request merged**